app/src/main/java/org/astraea/producer/Sender.java

```diff
@@ -23,6 +23,10 @@
 
   Sender<Key, Value> headers(Collection<Header> headers);
 
+  default Sender<Key, Value> transaction() {
```

Comment:
why this default implementation is required?

Reply:
This is used for identifying whether it is a transactional send.
As `run()` is called, we can choose the "transactional producer" or the "non-transactional producer".

Reply:
My point was - why we need this default implementation as there is only one sub class

Reply:
Initially, I want to keep the API unchanged. Because not all sender need to set transactional send.

Maybe it is also necessary for **not transactional sender** to implement **transaction()** method.
I will discard the `default` keyword.